src/storage/admin/StatisTask.cpp

```diff
@@ -0,0 +1,230 @@
+        tasks.emplace_back(std::move(task));
+    }
+    return tasks;
+}
+
+// Only statis the specified tags and edges
+kvstore::ResultCode
+StatisTask::genSubTask(GraphSpaceID spaceId,
+                       PartitionID part,
+                       std::vector<TagID> tags,
+                       std::vector<EdgeType> edges) {
+    auto vIdLenRet = env_->schemaMan_->getSpaceVidLen(spaceId);
+    if (!vIdLenRet.ok()) {
+        LOG(ERROR) << "Get space vid length failed";
+        return kvstore::ResultCode::ERR_SPACE_NOT_FOUND;
+    }
+    auto vIdLen = vIdLenRet.value();
+
+    LOG(INFO) << "Start statis task";
+    CHECK_NOTNULL(env_->kvstore_);
+    auto prefix = NebulaKeyUtils::partPrefix(part);
+    std::unique_ptr<kvstore::KVIterator> iter;
+    auto ret = env_->kvstore_->prefix(spaceId, part, prefix, &iter);
+    if (ret != kvstore::ResultCode::SUCCEEDED) {
+        LOG(ERROR) << "Statis task failed";
+        return ret;
+    } else {
+        LOG(INFO) << "Statis task successful";
+    }
+
+    std::unordered_map<TagID, int64_t>    tagsVertices;
+    std::unordered_map<EdgeType, int64_t> edgetypeEdges;
+    int64_t                               spaceVertices = 0;
+    int64_t                               spaceEdges = 0;
+
+    TagID                                 lastTagId = 0;
+    VertexID                              lastVertexId = "";
+
+    VertexID                              lastSrcVertexId = "";
+    EdgeType                              lastEdgeType = 0;
+    VertexID                              lastDstVertexId = "";
+    EdgeRanking                           lastRank = 0;
+
+    for (auto tag : tags) {
+        tagsVertices[tag] = 0;
+    }
+
+    for (auto edge : edges) {
+        edgetypeEdges[edge] = 0;
+    }
+
+    // Only statis valid vetex data
+    // For example
+    // Vid  tagId  version
+    // 1     1     2
+    // 1     1     1
+    // 2     2     2   (invalid data)
+    // 2     2     1   (invalid data)
+    // 2     3     1
+    // 3     1     2
+    while (iter && iter->valid()) {
+         auto key = iter->key();
+         if (NebulaKeyUtils::isVertex(vIdLen, key)) {
+             auto vId = NebulaKeyUtils::getVertexId(vIdLen, key).str();
+             auto tagId = NebulaKeyUtils::getTagId(vIdLen, key);
+
+            auto it = tagsVertices.find(tagId);
+            if (it == tagsVertices.end()) {
+                // Invalid data
+                iter->next();
+                continue;
+            }
+
+            if (vId == lastVertexId) {
+                if (tagId == lastTagId) {
+                    // Multi version
+                } else {
+                    tagsVertices[tagId] += 1;
+                    lastTagId = tagId;
+                }
+            } else {
+                tagsVertices[tagId] += 1;
+                spaceVertices++;
+                lastTagId = tagId;
+                lastVertexId  = vId;
+            }
```

Comment:
A question，Is remove duplicate vid for `spaceVertices` at here? for example : 
tag1 (c1)
tag2 (c1)

vertex1 : tag1(c1) ---> count(v) of tag1 == 1
vertex1: tag2(c1) ---> count(v) of tag2 == 1
so the spaceVertices == 1 ? or 2?

Reply:
Good question.
First of all, `spaceVertices` is 1 as in your case above.

Here use `lastVertexId` and current vid to remove duplicate data.
You can refer to the case on line 105.

Reply:
> Good question.
> First of all, `spaceVertices` is 1 as in your case above.
> 
> Here use `lastVertexId` and current vid to remove duplicate data.
> You can refer to the case on line 105.

got, nice.